File: 9-palindrome-number/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x == 0)return 1;
        else if (x < 0)return 0;
        long int resivedX = 0;
        int testX = x;
        while (testX) {
            resivedX = (resivedX * 10) + testX % 10;
            testX /= 10;
        }
        
        return resivedX == x ;
    }
};

File: 9-palindrome-number/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        string = str(x)
        i = 0 
        length = len(string) - 1
        while i < length - i:
            if string[i] != string[length - i]:
                return False
            i+=1
        
        return True